packages/react-dom/src/events/__tests__/DOMModernPluginEventSystem-test.internal.js

```diff
@@ -119,4 +119,33 @@ describe('DOMModernPluginEventSystem', () => {
     expect(log[4]).toEqual(['bubble', divElement]);
     expect(log[5]).toEqual(['bubble', buttonElement]);
   });
+
+  it('handle propagation of click events correctly with FB primer', () => {
+    ReactFeatureFlags.enableLegacyFBPrimerSupport = true;
+    const aRef = React.createRef();
+    const log = [];
+    // Stop propagation throught the React system
+    const onClick = jest.fn(e => e.stopPropagation());
+
+    function Test() {
+      return (
+        <a ref={aRef} href="#" onClick={onClick} rel="dialog">
+          Click me
+        </a>
+      );
+    }
+
+    ReactDOM.render(<Test />, container);
+
+    // Fake primer
+    document.addEventListener('click', e => {
+      if (e.target.rel === 'dialog') {
+        log.push('primer');
+      }
+    });
+    let aElement = aRef.current;
+    dispatchClickEvent(aElement);
+    expect(onClick).toHaveBeenCalledTimes(1);
+    expect(log).toEqual(['primer']);
```

Comment:
nit This test might be slightly more robust if it verified that another React-managed ancestor node didn't also get the click event. (As it stands, this test could pass if we just didn't stop propagating anything ever.)

We might also add a test for an anchor tag without one of our whitelisted rels.

Reply:
I've updated this test to what I think you meant. Is that okay now?

Reply:
Yeah. The extra React wrapper check looks good.